source/speech.py

```diff
@@ -27,6 +27,10 @@
 import speechDictHandler
 import characterProcessing
 import languageHandler
+import extensionPoints
+
+# inform those who want to know that there is new speech
+preSpeech = extensionPoints.Action()

```

Comment:
Also, this might have to be pre_speech according to the conventions.

Reply:
I think the conventions were discussed on PR #7606, though the underscores were dropped from that diff. I would be happy with the underscores and certainly want to keep it consistent, perhaps it should be made official on the wiki so that everyone is on the same page. @josephsl  would you be on board with this?

Reply:
There may be a minor performance impact here, very little compared to the rest of what happens before speech goes to the synth.

Which raises a bigger question about the approach of using  an extension point to gather the speech, should we be capturing the behaviour that happens between this line, and the eventual `getSynth().speak(speechSequence)`?

I think we probably should, otherwise there is no way for us to test things like auto language / dialect switching or symbol levels.

Perhaps, instead of an extension point for speech, the addon should install and select a dummy synthesizer.

Reply:
both could be useful actually. The synth captures the output and the extension point the first call.